tests/unit/amo/components/TestRatingManager.js

```diff
@@ -84,471 +103,321 @@ describe(__filename, () => {
 
-      const prompt = root.find('.RatingManager-legend').html();
-      expect(prompt).toContain('Are you sure you want to delete your review');
-      expect(prompt).toContain(addon.name);
+      // This verifies that AddonReviewManagerRating receives the expected
+      // rating from RatingManager while a deletion is happening.
+      expect(screen.getAllByTitle('Rated 3 out of 5')).toHaveLength(6);
     });
 
     it('prompts to delete a rating when beginningToDeleteReview', () => {
-      const addon = createInternalAddonWithLang({
-        ...fakeAddon,
-        name: createLocalizedString('uBlock Origin'),
+      renderWithReview({
+        review: {
+          ...fakeReview,
+          body: undefined,
+          user: { ...fakeReview.user, id: defaultUserId },
+        },
       });
-      const review = { ...fakeReview, body: undefined, score: 4 };
-      const { userId, store } = createStoreWithLatestReview({ addon, review });
-      store.dispatch(beginDeleteAddonReview({ reviewId: review.id }));
 
-      const root = render({ store, addon, userId });
+      userEvent.click(screen.getByRole('button', { name: 'Delete rating' }));
 
-      const prompt = root.find('.RatingManager-legend').html();
-      expect(prompt).toContain('Are you sure you want to delete your rating');
-      expect(prompt).toContain(addon.name);
+      expect(
+        screen.getByTextAcrossTags(
+          `Are you sure you want to delete your rating of ${defaultAddonName}?`,
+        ),
+      ).toBeInTheDocument();
     });
 
     it('still prompts to delete a review while deletingReview', () => {
-      const { addon, review, userId, store } = createStoreWithLatestReview();
-      store.dispatch(
-        deleteAddonReview({
-          addonId: addon.id,
-          errorHandlerId: 'some-error-handler',
-          reviewId: review.id,
-        }),
-      );
+      renderWithReview();
 
-      const root = render({ store, addon, userId });
+      userEvent.click(screen.getByRole('button', { name: 'Delete review' }));
 
-      const prompt = root.find('.RatingManager-legend').html();
-      expect(prompt).toContain('Are you sure you want to delete your review');
+      expect(
+        screen.getByTextAcrossTags(
+          `Are you sure you want to delete your review of ${defaultAddonName}?`,
+        ),
+      ).toBeInTheDocument();
     });
 
     it('shows AddonReviewCard with a saved review', () => {
-      const review = {
-        ...fakeReview,
-        body: 'This is hands down the best ad blocker',
-      };
-      const { addon, store } = createStoreWithLatestReview({ review });
-      const root = render({ addon, store });
-
-      const reviewCard = root.find(AddonReviewCard);
-      expect(reviewCard).toHaveProp('review', createInternalReview(review));
-    });
+      const body = 'This is some review copy';
+      renderWithReview({ review: { ...fakeReview, body } });
 
-    it('shows controls by default', () => {
```

Comment:
These tests for showing and hiding controls are covered by some tests above that exercise the component.